### PR TITLE
Introduce (reflective) Arb testing with Kotest

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: buildSrc/src/main/kotlin/ort-kotlin-conventions.gradle.kts [hunk 1]
```diff
@@ -61,6 +61,7 @@ testing {
                 implementation(project(":utils:test-utils"))
 
                 implementation(libs.kotest.assertions.core)
+                implementation(libs.kotest.property)
```
Comment (fviernau) [inline buildSrc/src/main/kotlin/ort-kotlin-conventions.gradle.kts:64]: commit-msg: convered -> converted

Comment (fviernau) [inline buildSrc/src/main/kotlin/ort-kotlin-conventions.gradle.kts:64]: just curious: What is the experiment after? What is the unkown and how can we make use of outcome?

I've a bit doubts that generating "meaningful" test data, is valuable for parameter values which do not matter. E.g. the need to emphasize (by using arbs) could also be a code smell hinting at suboptimal clarity.

Comment (sschuberth) [reply buildSrc/src/main/kotlin/ort-kotlin-conventions.gradle.kts:64]: > I've a bit doubts that generating "meaningful" test data, is valuable for parameter values which do not matter. E.g. the need to emphasize (by using arbs) could also be a code smell hinting at suboptimal clarity.

I see it pretty much as the opposite: Arb (which is short for "arbitrary") deliberately creates "unmeaningful" test data to emphasize that the test should work for any value, and not just for the hard-coded one. This also creates a chance to uncover edge-case bugs in tests where a test only works for an existing hard-coded value, whereas it should generally work. It's a little bit like [fuzzing](https://github.com/google/oss-fuzz).

So I see it as clarity being added by deliberately using arbitrary / random values.

> What is the experiment after? What is the unkown and how can we make use of outcome?

Eventually, I hope to get rid of e.g. some `EMPTY` constants that we have for many data classes. [That's because these constants are very often only used in tests](https://github.com/oss-review-toolkit/ort/pull/10575#discussion_r2340128908), and in places where the data should not actually be empty, but where we do not care about concrete values, so actually the data should be arbitrary.

File: plugins/package-managers/composer/src/funTest/kotlin/ComposerFunTest.kt [hunk 2]
```diff
@@ -39,7 +42,7 @@ import org.ossreviewtoolkit.utils.test.matchExpectedResult
 class ComposerFunTest : StringSpec({
     "Project files from vendor directories are ignored" {
         val projectFiles = ComposerFactory.create().mapDefinitionFiles(
-            File("."),
+            Arb.file().single(),
```
Comment (sschuberth) [inline plugins/package-managers/composer/src/funTest/kotlin/ComposerFunTest.kt:45]: @mnonnenmacher, I found a _little_ bit of a nicer syntax here.

File: buildSrc/src/main/kotlin/ort-kotlin-conventions.gradle.kts (hunk 1 above)
Comment (fviernau) [reply buildSrc/src/main/kotlin/ort-kotlin-conventions.gradle.kts:64]: LGTM, but let's get a second approval.

File: plugins/package-managers/composer/src/funTest/kotlin/ComposerFunTest.kt (hunk 2 above)
Comment (mnonnenmacher) [reply plugins/package-managers/composer/src/funTest/kotlin/ComposerFunTest.kt:45]: It's better, but if Kotest would provide a function like `arbitraryFile()` (and `arbitraryFiles()` for the not single case) it would make the tests more readable.
